container/container.go

```diff
@@ -463,6 +465,9 @@ func AttachStreams(ctx context.Context, streamConfig *runconfig.StreamConfig, op
 			}
 			<-done
 		}
+		if detached {
```

Comment:
Hmm, then the `detach` event may be lost if error happens. I can see that caller function doesn't care much about it's error, so I prefer not losing `detach` event to keeping error.